x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -319,6 +364,81 @@ export class EndpointDocGenerator {
     };
   }
 
+  private createNode(id: string, events: Event[]): TreeNode {
+    const relatedEvents: Event[] = [];
+    const lifecycle: Event[] = [];
+    events.forEach(event => {
+      if (event.event.category === 'process') {
+        lifecycle.push(event);
+      } else {
+        relatedEvents.push(event);
+      }
+    });
+    return { id, lifecycle, relatedEvents };
+  }
+
+  /**
+   * This generates a full resolver tree and keeps the entire tree in memory. This is useful for tests that want
+   * to compare results from elasticsearch with the actual events created by this generator. Because all the events
+   * are stored in memory do not use this function to generate large trees.
+   *
+   * @param options - options for the layout of the tree, like how many children, generations, and ancestry
+   * @returns a Tree structure that makes accessing specific events easier
+   */
+  public generateTree(options: TreeOptions = {}): Tree {
+    const getEntityID = (event: Event) => {
+      return event.process.entity_id;
+    };
+    const ancestry = this.createAlertEventAncestry(
+      options.ancestors,
+      options.relatedEvents,
+      options.percentWithRelated,
+      options.percentTerminated
+    );
+
+    const ancestryNodes: Map<string, TreeNode> = new Map();
```

Comment:
:grey_question: Purely a point of style here: It could read a little easier, perhaps, if you were to make this something like
`_.reduce(ancestry.slice(0,-1), (a,v)=>{ return a.set(getEntityId(v), v) }, new Map())`

Reply:
Good idea, I'll update it.